File: QF-Test/qftest-6.0.3/jython/Lib/ssh.py
from de.qfs.apps.qftest.shared.script.modules import SSH as _WrappedSSH
#  redirected constants
class Session(object):
    #  redirected constants
    #  redirected methods
    def __init__(self, username, server, port=22, password=None, keyfile=None, checkKnownHosts=True, knownHostsFile=None, connectTimeout=5000, command="bash", input=None, inputFile=None, mixOut=False, strong=False):
        """
        Creata a new Session.
        
        @param username
        The username to use.
        @param server
        The remote server to connect to.
        @param port
        The port to use for ssh login (optional, default 22).
        @param password
        The password to use (optional).
        @param keyfile
        The name of the file containing the key for public key
        authentication. Used only if no password is specified
        (optional, default is <user_home>/.ssh/id_rsa).
        @param checkKnownHost
        Whether to check the fingerprint of the remote server. For
        security reasons this should be enabled (optional, default
        true).
        @param knownHostFile
        A file containing the fingerprint for the remote server
        (optional, default is <user_home>/.ssh/known_hosts).
        @param connectTimeout
        Time in ms to wait for a connection. -1 to wait forever
        (optional, default is 5000).
        @param command
        The command to execute on the remote server (optional,
        default is 'bash').
        @param input
        Text to be bassed to the stdin of the executing program.
        Ignored if inputFile is specified.
        @param inputFile
        The path to a local file or nothing. If a file is
        specified, the content of this file is passed to the stdin
        of the executing program.
        @param mixOut
        If true, stdout and stderr of the executing program are
        captured together in the stdout stream with stderr
        remaining empty (optional, default false).
        @throws NoSuchAlgorithmException
        @throws JSchException
        @throws FileNotFoundException
        """
        self.__wrappedInstanceSession = _WrappedSSH.Session.instance(username, server, port, password, keyfile, checkKnownHosts, knownHostsFile, connectTimeout, command, input, inputFile, mixOut, strong)
    
    def setHostKeyChecking(self, check, knownHostsFile=None):
        """
        Define whether host key checking should take place.
        
        @param check
        Whether to check the fingerprint of the remote server.
        @param knownHostFile
        A file containing the fingerprint for the remote server
        (optional, default is <user_home>/.ssh/known_hosts).
        @throws JSchException
        """
        if check:
            check = True
        
        else:
            check = False
        
        self.__wrappedInstanceSession.setHostKeyChecking(check, knownHostsFile)
    
    def setConnectTimeout(self, timeout):
        """
        Set the connection timeout for the session.
        
        @param timeout
        Time in ms to wait for a connection. -1 to wait forever.
        """
        self.__wrappedInstanceSession.setConnectTimeout(timeout)
    
    def setCommand(self, command):
        """
        Set the command to execute.
        
        @param command
        The command to execute.
        """
        self.__wrappedInstanceSession.setCommand(command)
    
    def setInput(self, input):
        """
        Set the text to be passed to the stdin of the executing program.
        
        @param input
        Text to be bassed to the stdin of the executing program.
        Ignored if inputFile is also specified for this Session.
        """
        self.__wrappedInstanceSession.setInput(input)
    
    def setInputFile(self, inputFile):
        """
        Set the file with the text to be passed to the stdin of the executing
        program.
        
        @param inputFile
        The path to a local file. If a file is specified, the
        content of this file is passed to the stdin of the
        executing program.
        @throws FileNotFoundException
        """
        self.__wrappedInstanceSession.setInputFile(inputFile)
    
    def setEncoding(self, encoding):
        """
        Set the encoding for the stdout and stderr streams. Default is UTF-8.
        
        @param encoding
        The encoding to set. Must be a valid encoding name for the
        running JRE.
        """
        self.__wrappedInstanceSession.setEncoding(encoding)
    
    def open(self):
        """
        Open the session.
        
        @throws JSchException
        @throws IOException
        """
        self.__wrappedInstanceSession.open()
    
    def send(self, input, addNewline=True):
        """
        Send some input to the session.
        
        @param input
        The text to send.
        @param addNewline
        Whether to append a newline character if not already
        present (optional, default true).
        @throws IOException
        """
        if addNewline:
            addNewline = True
        
        else:
            addNewline = False
        
        self.__wrappedInstanceSession.send(input, addNewline)
    
    def expect(self, regexp, timeout=1000):
        """
        Wait for some output from the session. Only output after the last
        match for this method is taken into account.
        
        @param regexp
        A regular expression matching the text to wait for.
        @param timeout
        The maximum time to wait in ms. -1 to wait forever
        (optional, default 1000).
        
        @return The output from the process between the end of the previous
        match and the end of this match. null if no matchin output
        was received within the given time.
        @throws UnsupportedEncodingException
        @throws InterruptedException
        """
        return self.__wrappedInstanceSession.expect(regexp, timeout)
    
    def wait(self, timeout=1000):
        """
        Wait for the executing command to finish.
        
        @param timeout
        The maximum time to wait in ms. -1 to wait forever
        (optional, default 1000).
        @return
        @return
        
        @return true if the command did finish, false if it is still running.
        @throws InterruptedException
        """
        return self.__wrappedInstanceSession.waitTime(timeout)
    
    def waitTime(self, timeout=1000):
        """
        Wait for the executing command to finish.
        
        @param timeout
        The maximum time to wait in ms. -1 to wait forever
        (optional, default 1000).
        @return
        @return
        
        @return true if the command did finish, false if it is still running.
        @throws InterruptedException
        """
        return self.__wrappedInstanceSession.waitTime(timeout)
    
    def close(self):
        """
        Close the session.
        
        @throws IOException
        """
        self.__wrappedInstanceSession.close()
    
    def getExitCode(self):
        """
        Get the exit code of the command.
        
        @return The exit code of the command.
        """
        return self.__wrappedInstanceSession.getExitCode()
    
    def getStdout(self):
        """
        Get the output from the command to stdout.
        
        @return The output from the command to stdout. Inclues output to
        stderr if mixOut was set to true for this Session.
        @throws UnsupportedEncodingException
        """
        return self.__wrappedInstanceSession.getStdout()
    
    def getStderr(self):
        """
        Get the output from the command to stderr.
        
        @return The output from the command to stderr. Empty if mixOut was
        set to private Object stderr; true for this Session.
        @throws UnsupportedEncodingException
        """
        return self.__wrappedInstanceSession.getStderr()
    

File: QF-Test/qftest-6.0.3/javascript/ssh.js
exports.__esModule = true;
var module$input = exports;
var SSH = function() {
};
var _WrappedSSH = Java.type("de.qfs.apps.qftest.shared.script.modules.SSH");
SSH.Session = function() {
  _WrappedSSH.Session = Java.type("de.qfs.apps.qftest.shared.script.modules.SSH.Session");
  var __wrappedInstanceSession;
  function Session(username, server, port, password, keyfile, checkKnownHosts, knownHostsFile, connectTimeout, command, input, inputFile, mixOut, strong) {
    if (port === undefined) {
      port = 22;
    }
    if (password === undefined) {
      password = null;
    }
    if (keyfile === undefined) {
      keyfile = null;
    }
    if (checkKnownHosts === undefined) {
      checkKnownHosts = true;
    }
    if (knownHostsFile === undefined) {
      knownHostsFile = null;
    }
    if (connectTimeout === undefined) {
      connectTimeout = 5E3;
    }
    if (command === undefined) {
      command = "bash";
    }
    if (input === undefined) {
      input = null;
    }
    if (inputFile === undefined) {
      inputFile = null;
    }
    if (mixOut === undefined) {
      mixOut = false;
    }
    if (strong === undefined) {
      strong = false;
    }
    __wrappedInstanceSession = _WrappedSSH.Session.instance(username, server, port, password, keyfile, checkKnownHosts, knownHostsFile, connectTimeout, command, input, inputFile, mixOut, strong);
  }
  Session.prototype.setHostKeyChecking = parameterfy(function(check, knownHostsFile) {
    if (knownHostsFile === undefined) {
      knownHostsFile = null;
    }
    __wrappedInstanceSession.setHostKeyChecking(check, knownHostsFile);
  });
  Session.prototype.setConnectTimeout = parameterfy(function(timeout) {
    __wrappedInstanceSession.setConnectTimeout(timeout);
  });
  Session.prototype.setCommand = parameterfy(function(command) {
    __wrappedInstanceSession.setCommand(command);
  });
  Session.prototype.setInput = parameterfy(function(input) {
    __wrappedInstanceSession.setInput(input);
  });
  Session.prototype.setInputFile = parameterfy(function(inputFile) {
    __wrappedInstanceSession.setInputFile(inputFile);
  });
  Session.prototype.setEncoding = parameterfy(function(encoding) {
    __wrappedInstanceSession.setEncoding(encoding);
  });
  Session.prototype.open = parameterfy(function() {
    __wrappedInstanceSession.open();
  });
  Session.prototype.send = parameterfy(function(input, addNewline) {
    if (addNewline === undefined) {
      addNewline = true;
    }
    __wrappedInstanceSession.send(input, addNewline);
  });
  Session.prototype.expect = parameterfy(function(regexp, timeout) {
    if (timeout === undefined) {
      timeout = 1E3;
    }
    return __wrappedInstanceSession.expect(regexp, timeout);
  });
  Session.prototype.waitTime = parameterfy(function(timeout) {
    if (timeout === undefined) {
      timeout = 1E3;
    }
    return __wrappedInstanceSession.waitTime(timeout);
  });
  Session.prototype.close = parameterfy(function() {
    __wrappedInstanceSession.close();
  });
  Session.prototype.getExitCode = parameterfy(function() {
    return __wrappedInstanceSession.getExitCode();
  });
  Session.prototype.getStdout = parameterfy(function() {
    return __wrappedInstanceSession.getStdout();
  });
  Session.prototype.getStderr = parameterfy(function() {
    return __wrappedInstanceSession.getStderr();
  });
  return Session;
}();
var $jscompDefaultExport = SSH;
module$input.SSH = SSH;
module$input["default"] = $jscompDefaultExport;
